validator.py

```diff
@@ -1,23 +1,58 @@
 import sys
+from jsonschema.exceptions import RefResolutionError
+from colors import color
 from os import path, getcwd
-from openapi_spec_validator import validate_spec_url
+from openapi_spec_validator.handlers import UrlHandler
+from openapi_spec_validator import openapi_v3_spec_validator
 
 
 def validate(file_name):
-    validate_spec_url('file://' + path.join(getcwd(), file_name))
+    counter = 0
+    try:
+
+        handler = UrlHandler('file')
+        spec = handler('file://' + path.join(getcwd(), file_name))
+        errors_iterator = openapi_v3_spec_validator.iter_errors(spec)
+
+        for i in errors_iterator:
+            counter += 1
+            print_error(counter, ':'.join(i.absolute_path), i.message, i.instance)
+
+    except RefResolutionError as exception:
+        counter += 1
+        message = f'Unable to resolve reference {exception.__context__.args[0]} in {exception.args[0]}'
+        print_error(counter, '', message, '')
+    except:
+        counter += 1
+        print_error(counter, '', sys.exc_info()[0], '')
+    finally:
+        if counter > 0:
+            print()
+            print(color(' [FAIL] %d errors found ' % counter, fg='white', bg='red', style='bold'))
+            return 1
+        else:
+            print(color(' [PASS] No errors found ', fg='white', bg='green', style='bold'))
+            return 0
+
+
+def print_error(count, path, message, instance):
+    print()
+    print(color('Error #%d in [%s]:' % (count, path or 'unknown'), style='bold'))
+    print("    %s" % message)
+    print("    %s" % instance)
 
 
 def help():
-    print ('usage: ' + path.basename(__file__) + ' <spec>')
+    print('usage: ' + path.basename(__file__) + ' <spec>')
 
 
 def main(argv):
-    if (len(argv) != 1):
+    if len(argv) == 0:
```

Comment:
Not sure if we should do it but let's get this merged and we discuss afterwards (it's not really related to what we're doing now).

Reply:
This was done so I could actually run it in pycharm, since it appends `<host> <ip>` o something like that to all calls

Reply:
Ok, no worries. But maybe, you know, separate commits =)